pkg/virt-handler/device-manager/usb/manager.go

```diff
@@ -0,0 +1,449 @@
+package usb
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"kubevirt.io/api/usb/v1alpha1"
+	"kubevirt.io/client-go/log"
+)
+
+// The public facing API for virt-handler
+type USBManagerInterface interface {
+	Run(stopCh chan struct{})
+}
+
+// The handler to store and access Plugin's states
+type state struct {
+	// FIXME: Should the resourceName be unique across NodeConfigs?
```

Comment:
Can we have more than one instance of the CRD? For example, for KubeVirt CR we can only have a single one in the cluster

Reply:
First of all, I agree that is best to gather feedback on how to handle this part, to be able to come up with a API that can support multiple use cases.

> Can we have more than one instance of the CRD

Yes. The initial idea is to have each NodeConfig to bind to a resource name with one or more USB devices. When the resource name is attached to the VMI, all USB devices described by the NodeConfig should be provided to the VM.

I can better document this on the top. 

P.S: In relation to the FIXME comment: I think the resource name should be unique. If we have two different NodeConfig with the same `kubevirt.io/storage` resource name for different USB devices, we can't guarantee which device would be assign to each VM.  It would make more sense to have two different resource names, let's say, `storage-fast` and `storage-slow`. So, I put that `FIXME` there because we need to better handle this, but I wanted to make this PR and get the conversation started.

Reply:
@victortoso I think this will be properly fixed with [DRA](https://kubernetes.io/blog/2022/12/15/dynamic-resource-allocation/) where you can specify resource classes. 

The device plugin framework is quite simple and not very flexible. In general, it is almost impossible to guarantee the same device assignment if you have multiple devices under the same resource name. We have the same issue for NVMe devices.

So, if you want to support multiple instances of the CRDs to specify different properties of the device you could design the NodeConfig CRD in a way that could be reusable by DRA. 

What I mean by this, is that for DRA resource class can use any k8s object to configure the parameters. You can have a look at [my example](https://gist.github.com/alicefr/592591b18a99cf126dd82110d8fa74ea). From the example in the `ResourceClass` you have `parametersRef` this could reference the future of your `NodeConfig` (in my example I used a ConfigMap). Hence, different resource class will reference a different instance of the NodeConfig.

Reply:
However, what I'd avoid is having a CR per device or too many CRs. Otherwise, it becomes a nightmare to configure. 

I don't think you can really solve the problem that 2 (or more) devices can land under the same resource name without DRA and the `ResourceClaim`.
The only way for this is that you assign different resource names, like putting a number in the resource name, hence that you have a 1:1 relationship between the single device and the resource name. However, you risk to confusing the user that doesn't know which resource needs to reference.

Just an example, you might have `device1` and `device2` with the same product and vendor id, and you have assigned the resource name `testdevice.io/usb`. Either you expose on the node:
```
"testdevice.io/usb": 2 <- both devices under the same name
```
or 
```
"testdevice.io/usb1": 1 <- device1
"testdevice.io/usb2": 1 <- device2
```

Reply:
The alternative to this is keeping a single NodeConfig CR that will be unique in the cluster and specifying multiple devices and resources (as it is in the case of KubeVirt CR):

From your example:
```yaml
apiVersion: usb.kubevirt.io/v1alpha1
kind: NodeConfig
metadata:
  name: test
  namespace: default
spec:
  usb:
   - resourceName: kubevirt.io/storage-fast
     usbHostDevices:
      - selectByVendorProduct: 46f4:0001
   - resourceName: kubevirt.io/storage-slow
     usbHostDevices:
      - selectByVendorProduct: 46f4:0002
```

Reply:
@alicefr Can you clarify your concern with usability?

Generally, I don't think it would be much of a problem to have two CR requesting the same resource name. It is possible to implement an update of devices that a device plugin is serving.

Reply:
I don't have a particular concern, just how we want to design the configuration. What does the CR represent?
As I stated above, I think we can also have multiple CRs and IMO it could be a good preparation for DRA.

Possible issues you might encounter using multiple CRs are:
- reconciling multiple CRs, for example, if you declare the same vendor/product ID in multiple CRs under different resource names. If this is declared in a single one, then you can reject the update-
- from a usability perspective, if you end with too many objects, it can be more complex to manage, update and keep track of them. A single central place gives you an overview of all the USB devices.

Those concerns can be a workaround, but we should take a design decision about this CRD.

Reply:
I'm also hesitant about the number of CRs. I would expect a single list of devices mapped resources instead of multiple objects. 
I think this is just easy to manage from an admin's point of view. 
Also, I'd be concerned about the total number of objects.

Reply:
A single list would also be consistent with the permittedPCIDevices and MediatedDevices lists.